extensions/visual-studio-code-snippet-manager/src/components/Editor.tsx

```diff
@@ -0,0 +1,72 @@
+import { ActionPanel, Action, Form, useNavigation, showToast, Toast, popToRoot } from "@raycast/api";
+import { randomUUID } from "crypto";
+
+import fs from "fs";
+import { languages, path } from "../data";
+import Search from "./Search";
+
+const Editor = (props?: Item) => {
+  const { push } = useNavigation();
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            onSubmit={(v) => {
+              if (v.title === "" || v.prefix === "" || v.body === "") {
+                showToast({
+                  title: "error",
+                  message: "title, prefix and body are required",
+                  style: Toast.Style.Failure,
+                });
+              } else {
+                const data = {
+                  title: v.title,
+                  prefix: v.prefix,
+                  body: v.body.split("\n"),
+                  description: v.description,
+                  scope: v.language.join(),
+                };
+                if (fs.existsSync(path)) {
+                  const obj = JSON.parse(fs.readFileSync(path).toString());
+                  if (props && props.id) {
+                    obj[props.id] = data;
+                  } else {
+                    obj[`raycast_${randomUUID()}`] = data;
```

Comment:
Shouldn't we be using `title`? Otherwise `raycast_<uuid>` shows up in the suggestion list:

<img width="555" alt="image" src="https://user-images.githubusercontent.com/1155589/165081064-8ccbec82-117f-4d90-b08a-35767f010aad.png">

Reply:
(and if we do use title, we should check whether the same title is already present, and show a warning 😄 )

Reply:
Nice idea!
I will do that.